core/controllers/editor.py

```diff
@@ -101,9 +101,53 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+class EditorLogoutHandler(base.BaseHandler):
+    """Handles logout from editor page."""
+
+    @acl_decorators.can_play_exploration
+    def is_valid_exploration_id(self, exp_id):
+        """This is to check if exploration id is valid."""
+        pass
+
+    @acl_decorators.open_access
+    def get(self):
+        """Checks if exploration is published and redirects accordingly."""
+
+        return_url = str(self.request.get('return_url'))
+        if return_url:
```

Comment:
Slightly better to define url_to_redirect_to here (see below), do the if clause, and then do the self.redirect() at the very end (outside all the if/else). Then you don't need the else on line 129. Makes the code easier to understand.

Reply:
done.